vxquery-core/src/main/java/org/apache/vxquery/metadata/VXQueryCollectionOperatorDescriptor.java

```diff
@@ -88,25 +98,73 @@ public void open() throws HyracksDataException {
             public void nextFrame(ByteBuffer buffer) throws HyracksDataException {
                 fta.reset(buffer);
                 String collectionModifiedName = collectionName.replace("${nodeId}", nodeId);
+
                 File collectionDirectory = new File(collectionModifiedName);
+                //check if directory is in the local file system
+                if (collectionDirectory.exists()) {
+                    // Go through each tuple.
+                    if (collectionDirectory.isDirectory()) {
+                        for (int tupleIndex = 0; tupleIndex < fta.getTupleCount(); ++tupleIndex) {
+                            Iterator<File> it = FileUtils.iterateFiles(collectionDirectory, new VXQueryIOFileFilter(),
+                                    TrueFileFilter.INSTANCE);
+                            while (it.hasNext()) {
+                                File xmlDocument = it.next();
+                                if (LOGGER.isLoggable(Level.FINE)) {
+                                    LOGGER.fine("Starting to read XML document: " + xmlDocument.getAbsolutePath());
+                                }
+                                parser.parseElements(xmlDocument, writer, fta, tupleIndex);
+                            }
+                        }
+                    } else {
+                        throw new HyracksDataException("Invalid directory parameter (" + nodeId + ":"
+                                + collectionDirectory.getAbsolutePath() + ") passed to collection.");
+                    }
+                }
+                //else check in HDFS file system
+                else {
+                    //get instance of the HDFS filesystem
+                    HDFSFunctions hdfs = new HDFSFunctions();
+                    FileSystem fs = hdfs.getFileSystem();
+                    if (fs != null) {
+                        Path directory = new Path(collectionModifiedName);
```

Comment:
So this should also move at the open() method and just change every time in nextFrame().